File: backend/app/schemas/liquidaciones.py
from datetime import date, datetime

from pydantic import BaseModel, ConfigDict


class LiquidacionRead(BaseModel):
    model_config = ConfigDict(from_attributes=True)

    id: int
    numero_liquidacion: str
    personal_id: int
    periodo_mes: int
    periodo_anio: int
    fecha_generacion: date
    fecha_pago_programada: date
    total_entregas: float
    cantidad_entregas: int
    total_horas: float
    cantidad_horas: float
    total_labores: float
    cantidad_labores: int
    total_subsidio: float
    bonificaciones: float
    descuentos: float
    total_a_pagar: float
    estado: str
    fecha_pago_real: date | None
    metodo_pago: str
    referencia_pago: str | None
    observaciones: str | None
    fecha_creacion: datetime | None


class LiquidacionUpdate(BaseModel):
    bonificaciones: float | None = None
    descuentos: float | None = None
    observaciones: str | None = None
    fecha_pago_programada: date | None = None


class PagarLiquidacionRequest(BaseModel):
    fecha_pago_real: date
    metodo_pago: str = "transferencia"
    referencia_pago: str | None = None
    observaciones: str | None = None


class GenerarLiquidacionRequest(BaseModel):
    personal_id: int
    periodo_mes: int
    periodo_anio: int
    fecha_pago_programada: date
    bonificaciones: float = 0
    descuentos: float = 0
    observaciones: str | None = None


class ResumenPendientePago(BaseModel):
    personal_id: int
    codigo: str
    nombre_completo: str
    tipo_personal: str
    total_seriales: int
    total_mensajero: float
    total_horas: float
    total_horas_monto: float
    total_labores: int
    total_labores_monto: float
    total_subsidio: float
    total_pendiente: float
    ya_liquidado: bool

File: frontend/src/api/liquidaciones.ts
import api from "./client";

export interface Pendiente {
  personal_id: number; codigo: string; nombre_completo: string; tipo_personal: string;
  total_seriales: number; total_mensajero: number; total_horas: number; total_horas_monto: number;
  total_labores: number; total_labores_monto: number; total_subsidio: number;
  total_pendiente: number; ya_liquidado: boolean;
}

export interface Liquidacion {
  id: number; numero_liquidacion: string; personal_id: number;
  periodo_mes: number; periodo_anio: number; fecha_generacion: string;
  fecha_pago_programada: string; total_entregas: number; cantidad_entregas: number;
  total_horas: number; total_labores: number; total_subsidio: number;
  bonificaciones: number; descuentos: number;
  total_a_pagar: number; estado: string; fecha_pago_real: string | null;
  metodo_pago: string; referencia_pago: string | null; observaciones: string | null;
}

export const liqApi = {
  pendientes: (mes: number, anio: number) =>
    api.get<Pendiente[]>("/liquidaciones/pendientes", { params: { mes, anio } }),
  list: (params: object) => api.get<Liquidacion[]>("/liquidaciones/", { params }),
  generar: (data: object) => api.post<Liquidacion>("/liquidaciones/generar", data),
  aprobar: (id: number) => api.post<Liquidacion>(`/liquidaciones/${id}/aprobar`),
  pagar: (id: number, data: object) => api.post<Liquidacion>(`/liquidaciones/${id}/pagar`, data),
  delete: (id: number) => api.delete(`/liquidaciones/${id}`),
};
